src/s2_fos/interface.py

```diff
@@ -0,0 +1,254 @@
+    score: float = Field(description="Confidence scores for each field of study")
+
+    def to_dict(self):
+        return {self.label, self.score}
+
+
+class Prediction(BaseModel):
+    """
+    Describes the outcome of inference for one Instance
+
+    The fields below are examples only; please replace them with
+    appropriate fields for your model.
+
+    To learn more about declaring pydantic model fields, please see:
+    https://pydantic-docs.helpmanual.io/
+    """
+
+    field_of_studies_predicted_above_threshold: List[str] = Field(
+        description="Predicted fields of study for the paper"
+    )
+    scores: List[Score] = Field(description="Confidence scores for each field of study")
+
+
+class PredictorConfig(BaseSettings):
+    """
+    Configuration required by the model to do its work.
+    Uninitialized fields will be set via Environment variables.
+
+    The fields below are examples only; please replace them with ones
+    appropriate for your model. These serve as a record of the ENV
+    vars the consuming application needs to set.
+    """
+
+    # example_field: str = Field(default="asdf", description="Used to [...]")
+    thr_1: float = Field(default=0.552, description="Threshold for the first level")
+    thr_2: float = Field(default=0.621, description="Threshold for the second level")
+    thr_3: float = Field(default=0.7, description="Threshold for the second level")
+    thr_1_no_abstract: float = Field(default=0.621, description="Threshold for the first level no abstract")
+    thr_2_no_abstract: float = Field(default=0.655, description="Threshold for the second level no abstract")
+    thr_3_no_abstract: float = Field(default=0.7, description="Threshold for the third level no abstract")
+
+
+class S2FOS:
+    """
+    Interface on to your underlying model.
+
+    This class is instantiated at application startup as a singleton.
+    You should initialize your model inside of it, and implement
+    prediction methods.
+
+    If you specified an artifacts.tar.gz for your model, it will
+    have been extracted to `artifacts_dir`, provided as a constructor
+    arg below.
+    """
+
+    _config: PredictorConfig
+    data_dir: str
+
+    def __init__(self, data_dir: str):
+        self._config = PredictorConfig()
+        self.data_dir = data_dir
+        self._load_model()
+        self._model_lan_classifier = LanguageClassifier(data_dir=self.data_dir)
+
+    def _load_model(self) -> None:
+        """
+        Perform whatever start-up operations are required to get your
+        model ready for inference. This operation is performed only once
+        during the application life-cycle.
+        """
+        self._model = PredictProbabilities(model_path=self.data_dir)
+
+    def set_labels(self,
+                   threshold_list_np: np.array,
+                   abstract_np: np.array,
+                   thr_1_w_abstract: float = 0.52,
```

Comment:
Do you need to pass these in? They are not being used: they are being used from self._config anyway.